### PR TITLE
chore(lints): Add many rules for `ruff`

### PR DESCRIPTION
**Pull request by schloerke**
Made in separate PR to #1211 so that the file changes were separated

### CODE REVIEW

File: shiny/ui/_markdown.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def default_md_renderer(
         raise ModuleNotFoundError(
             "The default markdown parser requires the markdown-it-py package"
             " to be installed. Install it with `pip install markdown-it`."
-        )
+        ) from None
```
Comment (wch) [inline shiny/ui/_markdown.py:63]: Disable this lint and revert changes for now

File: shiny/ui/_modal.py [hunk 2]
```diff
@@ -129,16 +131,7 @@ def modal(
...
-        [
-            "if (window.bootstrap && !window.bootstrap.Modal.VERSION.match(/^4\\. /)) {",
-            "  var modal=new bootstrap.Modal(document.getElementById('shiny-modal'))",
-            "  modal.show()",
-            "} else {",
-            "  $('#shiny-modal').modal().focus()",
-            "}",
-        ]
-    )
+    js = "if (window.bootstrap && !window.bootstrap.Modal.VERSION.match(/^4\\. /)) {\n  var modal=new bootstrap.Modal(document.getElementById('shiny-modal'))\n  modal.show()\n} else {\n  $('#shiny-modal').modal().focus()\n}"
```
Comment (wch) [inline shiny/ui/_modal.py:134]: Convert to """.

Comment (wch) [reply shiny/ui/_modal.py:134]: And disable the linting rule

File: shiny/reactive/_reactives.py [hunk 3]
```diff
@@ -862,10 +862,10 @@ async def new_user_async_fn():
 
             return new_user_async_fn  # type: ignore
 
-        elif any([is_async_callable(arg) for arg in args]):
+        elif any(is_async_callable(arg) for arg in args):
```
[the remark this answers is not in the document]
Comment (schloerke) [reply shiny/reactive/_reactives.py:865]: `i for i in arr` returns a generator. When paired with `any()`, the generator is faster.

Rule with performance example: https://docs.astral.sh/ruff/rules/unnecessary-comprehension-any-all/

File: shiny/render/_try_render_plot.py [hunk 4]
```diff
@@ -137,8 +137,7 @@ def try_render_matplotlib(
         return (False, None)
 
     try:
-        import matplotlib
-        import matplotlib.pyplot as plt  # pyright: ignore[reportUnusedImport] # noqa: F401
+        import matplotlib.pyplot as plt  # pyright: ignore[reportUnusedImport]
```
[the remark this answers is not in the document]
Comment (schloerke) [reply shiny/render/_try_render_plot.py:140]: Even within this file we are not consistent with how matplotlib is being imported.  This lint made the usage consistent.

Keeping the lint.

**Pull request merged**